### PR TITLE
Eos 210 consumer read uncommitted

### LINKED ISSUE #210: Consumer "read_uncommitted" isolation level
Support the "read_uncommitted" isolation level for transactional consumers. Since the default behavior is to "read_committed", we'll add the isolation level config option in this task.

- [x] Allow configuring isolation level at consumer level
- [x] Provide isolation level to `Fetch` call
- [x] Provide isolation level to `ListOffsets` call
- [x] Ensure that we do not ignore aborted transactions

### PR DESCRIPTION
**Pull request by ianwsperber**
Resolves #210.

Add a `readUncommitted` flag to the consumer options which determines the isolation level for ListOffsets and Fetch requests. If we'd rather this option be `isolationLevel` or something of the sort that'd be fine as well.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -95,11 +98,17 @@ module.exports = class Client {
     retry,
     allowAutoTopicCreation,
     maxInFlightRequests,
+    readUncommitted = false,
   } = {}) {
+    const isolationLevel = readUncommitted
```
Comment (tulios) [inline src/index.js]: I think this is the best option right now

**Comment by tulios** [comment]
screen shot 2018-12-05 at 13 49 03

I think your test has a racing condition

**Comment by ianwsperber** [comment]
Thanks for catching that @tulios. It appears that the control record is not always in the fetch respond immediately after we've committed/aborted the transaction. I believe this is because the transaction coordinator will write the control record to the leaders during the `EndTxn` request, but does not wait for the record to be replicated. I'm less familiar with the nuances of leader/replication behavior than you likely are, but that's my intuition for what's happening. Anyways I moved the fetch & assertion inside a `retry` statement, with the idea that we should _eventually_ see the control record. I haven't seen the failure since.

**Comment by ianwsperber** [comment]
@tulios Can we merge this now?

**Pull request merged**